config/configauth/configauth.go

```diff
@@ -28,7 +29,19 @@ type Authentication struct {
 
 // GetServerAuthenticator attempts to select the appropriate auth.Server from the list of extensions,
 // based on the requested extension name. If an authenticator is not found, an error is returned.
-func (a Authentication) GetServerAuthenticator(extensions map[component.ID]component.Component) (auth.Server, error) {
+// Deprecated: [v0.97.0] Use GetServerAuthenticatorContext instead.
+func (a Authentication) GetServerAuthenticator(_ context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
```

Comment:
This function's signature should be left unchanged.  It should call `GetServerAuthenticatorContext`

Reply:
Really sorry, can you explain a bit further on what I am to do? @TylerHelmuth

Reply:
It should look something like

```
// Deprecated: [v0.97.0] Use GetServerAuthenticatorContext instead.
func (a Authentication) GetServerAuthenticator(extensions map[component.ID]component.Component) (auth.Server, error) {
  return a.GetServerAuthenticatorContext(context.Background(), extensions)
}

// put godoc commends here
func (a Authentication) GetServerAuthenticatorContext(_ context.Context, extensions map[component.ID]component.Component) (auth.Server, error) {
	if ext, found := extensions[a.AuthenticatorID]; found {
		if server, ok := ext.(auth.Server); ok {
			return server, nil
		}
		return nil, errNotServer
	}
	return nil, fmt.Errorf("failed to resolve authenticator %q: %w", a.AuthenticatorID, errAuthenticatorNotFound)
}
```

Reply:
I understand what to do now. 
Thank you for your help @TylerHelmuth